msal/src/Microsoft.Identity.Client/Platforms/net45/PlatformInformation.cs

```diff
@@ -57,6 +57,28 @@ public override string GetProductName()
             return "MSAL.Desktop";
         }
 
+        public override async Task<string> GetUserPrincipalNameAsync()
+        {
+            return await Task.Factory.StartNew(() =>
+            {
+                const int NameUserPrincipal = 8;
+                uint userNameSize = 0;
+                NativeMethods.GetUserNameEx(NameUserPrincipal, null, ref userNameSize);
+                if (userNameSize == 0)
+                {
+                    throw new MsalException(MsalError.GetUserNameFailed, MsalError.GetUserNameFailed, new Win32Exception(Marshal.GetLastWin32Error()));
```

Comment:
should this be MsalClientException ? ... same below?

Reply:
Was trying to mimic pre-existing ADAL behavior [here](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1153/files#diff-7b0de510ef9cb6269976fee57e0eaa1fR56) and [there](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1153/files#diff-7b0de510ef9cb6269976fee57e0eaa1fR62). But OK, I'm changing them to MsalClientException here and below in next commit.

---
In reply to: [207390781](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r207390781) [](ancestors = 207390781)